docs/dev/versions.md

```diff
@@ -0,0 +1,56 @@
+# Cirq version compatibility
+
+## Overview
+
+This document defines backwards compatibility guidelines across different versions of Cirq (either for code or data).
+
+
+## Semantic Versioning 2.0
+
+Cirq follows Semantic Versioning 2.0 ([semver](http://semver.org/)) for its public API. Each release version of Cirq has the form MAJOR.MINOR.PATCH. Changes to each number have the following meaning:
+
+
+
+*   **MAJOR**: Potentially backwards incompatible changes. Code and data that worked with a previous major release will not necessarily work with the new release. However, in most cases existing json serialized Cirq objects would be migratable to the newer release; see [Compatibility of serialized objects](#compatibility-of-serialized-objects) for details on data compatibility.
+*   **MINOR**: Backwards compatible features, speed improvements, etc. Code and data that worked with a previous minor release _and_ which depends only on the non-experimental public API will continue to work unchanged. For details on what is and is not the public API, see [What is covered](#what-is-covered).
+*   **PATCH**: Backwards compatible bug fixes.
+
+
+## What is covered
+
+Only the public APIs of Cirq are backwards compatible across minor and patch versions. The public APIs consist of
+
+
+
+*   All the documented [Python](https://quantumai.google/reference/python/cirq/all_symbols) functions and classes in the Cirq module and its submodules, except for
+    *   Private symbols: any function, class, etc. whose name start with **\_**
+    *   Experimental and `cirq.contrib` symbols, see [below](#what-is-not-covered) for details.
+*   Note that the code in the [examples/](https://github.com/quantumlib/Cirq/tree/master/examples) and [dev\_tools/](https://github.com/quantumlib/Cirq/tree/master/dev_tools) directories is not reachable through the `cirq` Python module and is thus not covered by the compatibility guarantee. Similarly, symbols in vendor packages, like [cirq-google](https://quantumai.google/reference/python/cirq_google/all_symbols), [cirq-aqt](https://quantumai.google/reference/python/cirq_aqt/all_symbols) are also not covered by the compatibility guarantee.
+    *   If a symbol is available through the `cirq` Python module or its submodules, but is not documented, then it is **not** considered part of the public API.
```

Comment:
what does it mean to be documented? have a docstring? be listed on the website?

Reply:
I mean having a docstring.